### PR TITLE
MULE-10295: target attribute accepts expressions

### PR DESCRIPTION
**Pull request by pablomusumeci**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java [hunk 1]
```diff
@@ -147,7 +151,23 @@ private ReturnDelegate createReturnDelegate() {
...
-    return StringUtils.isBlank(target) ? new ValueReturnDelegate(muleContext) : new TargetReturnDelegate(target, muleContext);
+    return !isTargetPresent() ? new ValueReturnDelegate(muleContext) : new TargetReturnDelegate(target, muleContext);
+  }
+
+  private boolean isTargetPresent() {
+    if (isBlank(target)) {
+      return false;
+    }
+
+    if (target.startsWith(FLOW_VARS)) {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java]: this is the wrong place to do this.. Application should not deploy. This should be validated on the OperationMessageProcessor constructor.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java [hunk 2]
```diff
@@ -147,7 +151,23 @@ private ReturnDelegate createReturnDelegate() {
...
+      throw new IllegalOperationException(format(INVALID_TARGET_MESSAGE, flowConstruct.getName(), operationModel.getName(),
+                                                 format("a flowVar with the '%s' prefix", FLOW_VARS)));
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java]: this message is redundant. Read it to yourself and you'll notice that as a user you wouldn't like to receive that message

File: modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessorTestCase.java [hunk 3]
```diff
@@ -41,6 +47,11 @@
 @RunWith(MockitoJUnitRunner.class)
 public class OperationMessageProcessorTestCase extends AbstractOperationMessageProcessorTestCase {
 
+  public static final String INVALID_TARGET =
```
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessorTestCase.java]: this should be the same constant than the validation uses. Make the original constant package level and place this test in the same package

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java]: or you could also validate on the initialise phase

Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java]: better to do it in the initialise

Comment (pablomusumeci) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java]: This validation runs when the return delegate is created, which happens in the `doInitialise` method.

**Pull request merged**